target/linux/ipq806x/image/Makefile

```diff
@@ -29,7 +29,6 @@ define Device/Default
 	KERNEL_INITRAMFS_PREFIX := $$(IMG_PREFIX)-$(1)-initramfs
 	KERNEL_PREFIX := $$(IMAGE_PREFIX)
 	KERNEL_LOADADDR = 0x42208000
-	SUPPORTED_DEVICES = $$(BOARD_NAME)
```

Comment:
Please use `SUPPORTED_DEVICES := $(subst _,$(comma),$(1))` here. It will use the part after the `define Device/`as "default" and replaces underline by comma to match the userspace boardname. It should match the userspace boardname afterwards.

For devices which are supported by lede-17.01, like the TP-Link Archer C2600, append the former used boardname in the device define:

```
define Device/tplink_c2600
	...
	SUPPORTED_DEVICES += c2600
	...
endef
```

I'm fine to kind of (soft) break compatibility for Devices which are only supported by master. They will be upgradeable by using `sysupgrade --force /path/to/sysupgrade.bin`.

Reply:
Ok, will do.